### PR TITLE
ethos classification

### PR DESCRIPTION
**Pull request by naamaz**

### CODE REVIEW

File: prepare/cards/ethos.py [hunk 1]
```diff
@@ -35,6 +35,16 @@
                 output_format="{label}",
                 postprocessors=["processors.take_first_non_empty_line"],
             ),
+            InputOutputTemplate(
+                input_format="Given this sentence: {text}. Classify if it contains hatespeech. Choices: {choices}. I would classify this sentence as: ",
```
Comment (yoavkatz) [inline prepare/cards/ethos.py:39]: It's "hate speech" more comon than "hatespeech"? Or is this the expected output?

Comment (naamaz) [reply prepare/cards/ethos.py:39]: These were the original choices and therefore this is the expected output. It works this way. I can try "hatespeech" instead and check. should work as well

**Review by yoavkatz — CHANGES_REQUESTED** [review]
Overall looks good.  Minor changes request.

File: prepare/cards/ethos.py (hunk 1 above)
Comment (yoavkatz) [reply prepare/cards/ethos.py:39]: The choices are "hate speech" and "not hate speech" so why are the instructions "Classify if it contains hatespeech"? (without a space).  Maybe it does not make a difference, but it seems wrong.

**Review by yoavkatz — APPROVED** [review]
I approve, but (1) I think you can still call it   "processors.hate_or_not"  or even "processors.hate_speech_or_not_hate_speech" (and" not string_or_not_hate", which is not clear)

File: src/unitxt/processors.py [hunk 2]
```diff
@@ -74,3 +74,22 @@ def process(self, instance):
...
+    def process(self, instance):
+        non_empty_line = instance.lower()
+        match = re.search(r"[.,!?;]", non_empty_line)
+        if match:
+            # Extract text up to the first punctuation
+            non_empty_line = non_empty_line[: match.start()]
+        return non_empty_line
+
+
+class HateOrNot(BaseFieldOperator):
```
[the remark this answers is not in the document]
Comment (naamaz) [reply src/unitxt/processors.py]: Done. Chose "hate speech" since it is not common to use "hatespeech" as far as I could see.

**Review by yoavkatz — APPROVED** [review]

**Pull request merged**